### ISSUE TITLE
Helm3: Upgrade with conditional dependencies not working

### ISSUE THREAD

**Issue by karuppiah7890** [helm/helm#6530]
Related issue - #6313 (helm v2 bug)

This happens only while using `--reuse-values`. When explicitly mentioning values yamls using `--values` it doesn't happen

Output of `helm version`:
```
version.BuildInfo{Version:"v3.0+unreleased", GitCommit:"3fa4465eec27e1a008189a3c22d7367a75cac136", GitTreeState:"clean", GoVersion:"go1.13.1"}
```

**Comment by karuppiah7890** [comment]
My guess is, this happens because we process the charts and the dependencies here

https://github.com/helm/helm/blob/eac6a60001934ef93517f52b2a87e04273da6e30/pkg/action/upgrade.go#L71

And then prepare upgrade here

https://github.com/helm/helm/blob/eac6a60001934ef93517f52b2a87e04273da6e30/pkg/action/upgrade.go#L83

And only during prepare upgrade we check the values present in the existing release to get the values to reuse which is here

https://github.com/helm/helm/blob/eac6a60001934ef93517f52b2a87e04273da6e30/pkg/action/upgrade.go#L138

**Comment by karuppiah7890** [comment]
This can be closed as #6556 has been merged
